platform/src/main/java/org/stellar/anchor/platform/configurator/ConfigManager.java

```diff
@@ -0,0 +1,192 @@
+package org.stellar.anchor.platform.configurator;
+
+import static org.stellar.anchor.api.platform.HealthCheckStatus.GREEN;
+import static org.stellar.anchor.platform.configurator.ConfigHelper.*;
+import static org.stellar.anchor.platform.configurator.ConfigMap.ConfigSource.FILE;
+import static org.stellar.anchor.util.Log.*;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.List;
+import java.util.Properties;
+import lombok.Builder;
+import lombok.Data;
+import lombok.SneakyThrows;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.core.env.PropertiesPropertySource;
+import org.springframework.core.io.Resource;
+import org.stellar.anchor.api.exception.InvalidConfigException;
+import org.stellar.anchor.api.platform.HealthCheckResult;
+import org.stellar.anchor.healthcheck.HealthCheckable;
+import org.stellar.anchor.util.GsonUtils;
+import org.stellar.anchor.util.Log;
+
+public class ConfigManager
+    implements ApplicationContextInitializer<ConfigurableApplicationContext>, HealthCheckable {
+
+  static String STELLAR_ANCHOR_CONFIG = "STELLAR_ANCHOR_CONFIG";
+  static ConfigManager configManager = new ConfigManager();
+
+  ConfigMap configMap;
+
+  private ConfigManager() {}
+
+  public static ConfigManager getInstance() {
+    return configManager;
+  }
+
+  @SneakyThrows
+  @Override
+  public void initialize(@NotNull ConfigurableApplicationContext applicationContext) {
+    // Read configuration from system environment variables, configuration file, and default values
+    info("Read and process configurations");
+    configMap = processConfigurations(applicationContext);
+
+    // Make sure no secret is leaked.
+    sanitize(configMap);
+
+    // Send values to Spring
+    sendToSpring(
+        applicationContext, configMap, List.of(new LogConfigAdapter(), new DataConfigAdapter()));
+  }
+
+  void sanitize(ConfigMap configMap) {
+    SecretManager.getInstance()
+        .secretVars
+        .forEach(
+            var -> {
+              if (configMap.get(var) != null) {
+                Log.warnF(
+                    "Possible secret leak of config[{}]. Please remove the secret from configuration.",
+                    var);
+                configMap.remove(var);
+              }
+            });
+  }
+
+  void sendToSpring(
+      ConfigurableApplicationContext applicationContext,
+      ConfigMap config,
+      List<SpringConfigAdapter> adapters)
+      throws InvalidConfigException {
+    Properties props = new Properties();
+    props.putAll(config.map());
+
+    // Set Platform configurations
+    PropertiesPropertySource pps = new PropertiesPropertySource("config", props);
```

Comment:
TBH, I am not a fan of using constants to replace all literals unless it brings value. In this case, it is not clear how much value does changing to constants bring but giving the meaning of the literal. Since it is not clear to me and you had mentioned it, so I had changed it to a constant.